test/let_value_with_stop_token_test.cpp

```diff
@@ -97,6 +98,40 @@ TEST_F(LetWithStopToken, Simple) {
   EXPECT_EQ(external_context, 42);
 }
 
+TEST_F(LetWithStopToken, InplaceStoppableStopSourceNoexcept) {
```

Comment:
Can you add a test for a sender that isn't noexcept qualified? It'd be good to ensure that let_value_with_stop_source isn't noexcept qualified when the lambda isn't noexcept qualified.